docs/decisions/0004-eventstore-abstraction.md

```
@@ -0,0 +1,308 @@
+   - `BatchValidation.Spec`
+   - `LocalPositionStamping.Spec`
+
+2. Implement the `EventStoreConfig` type class for its configuration type.
+
+3. Ensure events are JSON-serializable (`ToJSON`/`FromJSON` constraints).
+
+4. Handle toxic events (undecodable events) gracefully by returning `ToxicStreamEvent`/`ToxicAllEvent` messages rather than failing.
+
+## References
+
+- Event Sourcing pattern: https://martinfowler.com/eaaDev/EventSourcing.html
```

markdownlint-cli2 (0.18.1)

Bare URL used

(MD034, no-bare-urls)